numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && len == 7 && !strncmp(str, "generic", 7)) {
```

Comment:
Yes, but even bug fixes are not typically backported unless they are regressions or relatively severe, a "it never worked for 10 years and errors out" bug does not fall into that category most likely.